src/openttd.cpp

```diff
@@ -603,6 +614,42 @@ int openttd_main(int argc, char *argv[])
 		case 'P':
 			scanner->join_company_password = mgo.opt;
 			break;
+		case 'U': {
+			char* p = mgo.opt;
+
+			const char* protocol = "openttd://";
+			if (strncmp(p, protocol, strlen(protocol)) != 0) {
+				fprintf(stderr, "Unknown protocol\n");
+				break;
+			}
+
+			p += strlen(protocol);
+
+			char* network_conn = p;
+			p = strchr(p, '?');
+			*p = '\0';
+			p++;
+			scanner->network_conn = network_conn;
+
+			const char* name = nullptr;
+			const char* value = nullptr;
+			while (ParseQueryString(&name, &value, &p)) {
+				if (strcmp(name, "password") == 0) {
+					scanner->join_server_password = value;
+				} else if (strcmp(name, "join") == 0) {
+					scanner->join_company = value;
+				} else if (strcmp(name, "companypassword") == 0) {
+					scanner->join_company_password = value;
+				} else if (strcmp(name, "version") == 0) {
+					/* Nothing for now */
+				} else {
+					fprintf(stderr, "Unknown parameter: ");
+					fprintf(stderr, name);
+					fprintf(stderr, "\n");
```

Comment:
Use format strings here, you're passing user data (the `name` variable) directly in as the format string. This is a potential security vulnerability.
`fprintf(stderr, "Unknown parameter: %s\n", name);`

Reply:
Yeah, these kind of protocol handlers need to be safe with potentially malicious URLs triggered from untrusted websites.

Reply:
Huh, had no idea about that, thanks

Reply:
Quite apart from the security implications, one fprintf call is better than 3!

Reply:
True